core/src/main/java/org/mule/runtime/core/api/processor/strategy/ProcessingStrategyFactory.java

```diff
@@ -19,4 +19,12 @@
    * @return a new {@link ProcessingStrategy}.
    */
   ProcessingStrategy create(MuleContext muleContext, String schedulersNamePrefix);
+
+  /**
+   * @return a reference to the concrete implememntation of the {@link ProcessingStrategy} this facotry will create.
+   */
+  default Class<? extends ProcessingStrategy> getProcessingStrategyClass() {
```

Comment:
- Minor: but `getProcessingStrategyType` would be slight better.
- Isn't it better to force implementations to implement this, given it's used in conditions in Flow/Pipeline?